### ISSUE TITLE
Error: Cannot find module 'react' After adding a plugin

### ISSUE THREAD

**Issue by hhimanshu** [gatsbyjs/gatsby#18841]
## Description

I am following https://www.gatsbyjs.org/tutorial/part-three/ and the project fails with 

```sh
 Error: Cannot find module 'react'

  - loader.js:668 Function.Module._resolveFilename
    internal/modules/cjs/loader.js:668:15

  - loader.js:591 Function.Module._load
    internal/modules/cjs/loader.js:591:27
```

### Steps to reproduce
The video to reproduce is here

https://imgur.com/download/ugF3gu0

### Expected result

What should happen?

### Actual result

What happened.

### Environment
```sh
gatsby info --clipboard

  System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.14.0 - ~/.nvm/versions/node/v11.14.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.10.3 - ~/.nvm/versions/node/v11.14.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 77.0.3865.120
    Safari: 12.1.2
  npmPackages:
    gatsby: ^2.16.1 => 2.16.1
    gatsby-plugin-typography: ^2.3.14 => 2.3.14
  npmGlobalPackages:
    gatsby-cli: 2.7.30
```
Run `gatsby info --clipboard` in your project directory and paste the output here.

**Comment by KyleAMathews** [comment]
Can you remove your lock file and the node_modules directory and reinstall? Sometimes packages will get deleted like it looks react did.

**Comment by hhimanshu** [comment]
@KyleAMathews, after deleting the folder and running `yarn`, things work, but this raises a concern - Do we have to perform this step every single time when we add a new plugin. I tried with 2-3 different plugins and the failure steps were reproducible.

Coming from CRA (create-react-app), this is a bad user experience.

**Comment by AlbertVilaCalvo** [comment]
Got the exact same error following the same tutorial https://www.gatsbyjs.org/tutorial/part-three/ when running `gatsby develop`.

Then I did run `npm install` and got another error:
`Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons: ...`

Then I removed the `node_modules` and `package-lock.json` as suggested by @KyleAMathews  and it did work :)

Would be nice to get rid of this error, or to add a note with the fix in the tutorial.
